File: Q36.py
#Q36.py
class Solution:
    def isValidSudoku(self, b):
        row = [[0]*9 for _ in range(9)]
        col = [[0]*9 for _ in range(9)]
        box = [[0]*9 for _ in range(9)]
        for i in range(9):
            for j in range(9):
                if b[i][j] == '.': continue
                v = ord(b[i][j]) - 49
                k = (i//3)*3 + j//3
                if row[i][v] or col[j][v] or box[k][v]:
                    return False
                row[i][v] = col[j][v] = box[k][v] = 1
        return True

File: Q36.cpp
//Q36.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& b) {
        bool row[9][9] = {}, col[9][9] = {}, box[9][9] = {};
        for(int i=0;i<9;i++){
            for(int j=0;j<9;j++){
                if(b[i][j]=='.') continue;
                int v = b[i][j]-'1';
                int k = (i/3)*3 + j/3;
                if(row[i][v] || col[j][v] || box[k][v]) return false;
                row[i][v] = col[j][v] = box[k][v] = true;
            }
        }
        return true;
    }
};
